### PR TITLE
Show activity on sidebar

### PR DESCRIPTION
**Pull request by fonurr**

### CODE REVIEW

File: src/sidebar/activity/data/ActivityStore.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        const barrels = getSavedBarrels();
+        const projects = barrels.filter(barrel => barrel.type === BarrelType.Project);
+        const results = await Promise.all([
+            ...projects.map(({ id }) => ScreensStoreProvider.get(id).get().then(this.convertResult)),
+            ...barrels.map(({ id, type }) => new ZeplinComponentsStore(id, type).get().then(this.convertResult))
+        ]);
+
+        return {
+            data: flatten(results.map(result => result.data ?? [])),
+            errors: flatten(results.map(result => result.errors ?? []))
```
Comment (mertkahyaoglu) [inline src/sidebar/activity/data/ActivityStore.ts:23]: Does `flatMap` work here?

File: src/sidebar/activity/util/activityUtil.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import localization from "../../../localization";
+
+const MS_IN_MINUTE = 1000 * 60; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const MS_IN_HOUR = 60 * MS_IN_MINUTE; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const MS_IN_DAY = 24 * MS_IN_HOUR; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const MS_IN_WEEK = 7 * MS_IN_DAY; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const MS_IN_MONTH = 30 * MS_IN_DAY; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const MS_IN_YEAR = 365 * MS_IN_DAY; // eslint-disable-line @typescript-eslint/no-magic-numbers
+const AGO_YEAR_LIMIT = 2;
+
```
Comment (mertkahyaoglu) [inline src/sidebar/activity/util/activityUtil.ts:11]: Didn't test but this should work.
```suggestion
// eslint-disable @typescript-eslint/no-magic-numbers
const MS_IN_MINUTE = 1000 * 60;
const MS_IN_HOUR = 60 * MS_IN_MINUTE;
const MS_IN_DAY = 24 * MS_IN_HOUR;
const MS_IN_WEEK = 7 * MS_IN_DAY;
const MS_IN_MONTH = 30 * MS_IN_DAY;
const MS_IN_YEAR = 365 * MS_IN_DAY;
const AGO_YEAR_LIMIT = 2;
// eslint-enable @typescript-eslint/no-magic-numbers
```

File: src/sidebar/activity/data/ActivityStore.ts (hunk 1 above)
Comment (fonurr) [reply src/sidebar/activity/data/ActivityStore.ts:23]: yeah but, I wrote this custom flatten method which has empty checks at all levels, works better for our use case

Comment (mertkahyaoglu) [reply src/sidebar/activity/data/ActivityStore.ts:23]: Ahh ok. I remembered that now.

**Review by konukselim — APPROVED** [review]

**Pull request merged**